bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -97,96 +100,164 @@ protected NAWelcomeHome updateReadings() {
-                        : UnDefType.UNDEF;
+                return lastEvent.map(e -> toOnOffType(e.getIsArrival())).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_MESSAGE:
-                return lastEvent != null && lastEvent.getMessage() != null
-                        ? new StringType(lastEvent.getMessage().replace("<b>", "").replace("</b>", ""))
-                        : UnDefType.UNDEF;
+                return findEventMessage().map(
+                        m -> (State)new StringType(m.replace("<b>", "").replace("</b>", ""))
+                ).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_SUBTYPE:
-                return lastEvent != null ? toDecimalType(lastEvent.getSubType()) : UnDefType.UNDEF;
+                return lastEvent.map(e -> toDecimalType(e.getSubType())).orElse(UnDefType.UNDEF);
         }
         return super.getNAThingProperty(channelId);
     }
 
+    @Override
+    protected void triggerChannelIfRequired(String channelId) {
+        if(isNewLastEvent) {
+            if(CHANNEL_CAMERA_EVENT.equals(channelId)) {
+                Set<String> detectedObjectTypes = findDetectedObjectTypes(lastEvent);
+                for(String detectedType: detectedObjectTypes) {
+                    triggerChannel(channelId, detectedType);
+                }
+            }
+        }
+        super.triggerChannelIfRequired(channelId);
+    }
+
+    private static Set<String> findDetectedObjectTypes(Optional<NAWelcomeEvent> eventOptional) {
+        Set<String> detectedObjectTypes = new TreeSet<>();
+        if(!eventOptional.isPresent()) {
+            return detectedObjectTypes;
+        }
+
+        NAWelcomeEvent event = eventOptional.get();
+
+        if(event.getPersonId() != null) {
+            detectedObjectTypes.add(CAMERA_EVENT_OPTION_HUMAN_DETECTED);
+        }
+
+        if(NAWebhookCameraEvent.EventTypeEnum.MOVEMENT.toString().equals(event.getType())) {
+            detectedObjectTypes.add(CAMERA_EVENT_OPTION_MOVEMENT_DETECTED);
+        }
+
+        List<NAWelcomeSubEvent> subEvents = event.getEventList();
+        for(NAWelcomeSubEvent subEvent: subEvents) {
+            String detectedObjectType = translateDetectedObjectType(subEvent.getType());
+            detectedObjectTypes.add(detectedObjectType);
+        }
+        return detectedObjectTypes;
+    }
+
+    private static String translateDetectedObjectType(NAWelcomeSubEvent.TypeEnum type) {
+        switch (type) {
+            case HUMAN: return CAMERA_EVENT_OPTION_HUMAN_DETECTED;
+            case ANIMAL: return CAMERA_EVENT_OPTION_ANIMAL_DETECTED;
+            case VEHICLE: return CAMERA_EVENT_OPTION_VEHICLE_DETECTED;
+            default: throw new IllegalArgumentException("Unknown detected object type!");
+        }
+    }
+
+    private Optional<String> findEventMessage() {
+        if(lastEvent.isPresent()) {
+            @Nullable String message = lastEvent.get().getMessage();
```

Comment:
Is @Nullable required here ?

Reply:
yes, the variable can be NULL (it is not NonNull)